### PR TITLE
fix(ca1): scrape new endpoint

### PR DESCRIPTION
**Pull request by grossir**
Solves #1017

Changes:
- Target new endpoint
- Refactor to OpinionSiteLinear
- Implement dynamic backscraper to solve most recent gap

### CODE REVIEW

File: juriscraper/opinions/united_states/federal_appellate/ca1.py [hunk 1]
```diff
@@ -1,124 +1,104 @@
...
+    def _process_html(self):
+        for row in self.html.xpath("//tr[position() > 1]"):
+            title = row.xpath("td[2]/a/text()")[0]
+            url = row.xpath("td[2]/a/@href")[0]
+            status = self.get_status_from_opinion_title(title)
+            if status == "Errata":
+                logger.info(
+                    "Skipping Errata file. title: %s URL: %s", title, url
+                )
+                continue
```
Comment (flooie) [inline juriscraper/opinions/united_states/federal_appellate/ca1.py:31]: did we skip errata before? why the change?

Comment (grossir) [reply juriscraper/opinions/united_states/federal_appellate/ca1.py:31]: We didn't. Will correct this, just saw that we actually have a Errata status

Comment (flooie) [reply juriscraper/opinions/united_states/federal_appellate/ca1.py]: thanks.

File: juriscraper/opinions/united_states/federal_appellate/ca1.py [hunk 2]
```diff
@@ -1,124 +1,98 @@
...
-        lower_courts = []
-        for e in self.html.xpath("//tr[position() > 1]/td[4]/font"):
-            try:
-                lower_courts.append(e.xpath("./text()")[0].strip())
-            except IndexError:
-                lower_courts.append("")
-        return lower_courts
+        :return: status string
+        """
+        status = ""
```
Comment (flooie) [inline juriscraper/opinions/united_states/federal_appellate/ca1.py:49]: this line is unused

**Comment by flooie** [comment]
one minor thing but otherwise I think good

File: juriscraper/opinions/united_states/federal_appellate/ca1.py (hunk 2 above)
Comment (grossir) [reply juriscraper/opinions/united_states/federal_appellate/ca1.py]: Good eye, Just updated the PR removing this

**Review by quevon24 — COMMENTED** [review]
It works perfectly, the only detail I found is that it skips the first row of data. There are 20 items in https://www.ca1.uscourts.gov/opn/aci but it only returns 19

The first case is "US v. Polaco-Hance" but the extracted items begin with: "US v. Marte Carmona" which is the second item

**Review by quevon24 — CHANGES_REQUESTED** [review]
It works perfectly, the only detail I found is that it skips the first row of data. There are 20 items in https://www.ca1.uscourts.gov/opn/aci but it only returns 19

The first case is "US v. Polaco-Hance" but the extracted items begin with: "US v. Marte Carmona" which is the second item

**Comment by grossir** [comment]
Thanks @quevon24 I have corrected the xpath bug, please check again

**Review by quevon24 — APPROVED** [review]
all good 👍

I leave this message as a note. I had to force the installation of setuptools in the workflow so that the tests with python 3.12 would pass correctly. There is already an issue open here: https://github.com/actions/runner-images/issues/10021
